src/adr/0018-jmap-new-specs.md

```diff
@@ -0,0 +1,48 @@
+# 18. New JMAP specifications adoption
+
+Date: 2020-02-06
+
+## Status
+
+Proposed
+
+## Context
+
+Historically, James is an early adopter for the JMAP specification, and a first partial implementation was conducted when JMAP was just a draft. But with time, the IETF draft went with radical changes and the community could not keep this implementation up to date with the spec changes.
+
+As off summer 2019, JMAP core ([RFC 8620](https://tools.ietf.org/html/rfc8620)) and JMAP mail ([RFC 8621](https://tools.ietf.org/html/rfc8621)) have been officially published. Thus we should implement these new specifications.
+
+We need to keep in mind though that part of the community actively relies on the actual 'draft' implementation of JMAP existing in James. We should ensure no changes is done to that 'draft' protocol is done while implementing the new one.
+
+## Decision
+
+The proposed approach would be as follow:
+
+* Rename packages `server/protocols/jmap*` and guice packages `server/container/guice/protocols/jmap*` to `jmap-draft`. `JMAPServer` should also be renamed to `JMAPDraftServer` (this has already been contributed [here](https://github.com/apache/james-project/pull/164), thanks to @cketti). No modification in `jmap-draft` integration tests should be conducted afterwards.
+* Create a new `jmap` package, binding it on a new port.
+* Implement the new JMAP request structure with the echo method
+* Implement authentication and session of the new JMAP protocol
+* Implement protocol-level error handling
+* Port existing mailbox methods of `jmap-draft` to `jmap`
```

Comment:
Do you mean "duplicate" here?

Reply:
By "Port" I mean duplicate and adapt to the new specs. Should it be more explicit?

Reply:
Port can be understood as "adapt" without "duplicate".

Reply:
I thought with saying "methods of `jmap-draft` to `jmap`", and also with the stated fact that we don't modify `jmap-draft`, it was clear enough. But I can make it more explicit if you want

Reply:
You can still have 02 `jmap-rfc-8620` & jmap-draft distinct project and a "meta" `jmap` project deciding which servlets to be calling based on the Accept header version, no?